BASE/src/Microsoft.ApplicationInsights/Metrics/Implementation/AutocollectedMetricsExtraction/TraceMetricsExtractor.cs

```diff
@@ -0,0 +1,165 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Metrics
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Metrics;
+    using static System.FormattableString;
+
+    /// <summary>
+    /// An instance of this class is contained within the <see cref="AutocollectedMetricsExtractor"/> telemetry processor.
+    /// It extracts auto-collected, pre-aggregated (aka. "standard") metrics from TraceTelemetry objects which represent
+    /// invocations of the monitored service.
+    /// </summary>
+    internal class TraceMetricsExtractor : ISpecificAutocollectedMetricsExtractor
+    {
+        /// <summary>
+        /// The default value for the <see cref="MaxTraceSeverityLevelValuesToDiscover"/> property.        
+        /// </summary>
+        public const int MaxTraceSeverityLevelValuesToDiscoverDefault = 5;
+
+        /// <summary>
+        /// The default value for the <see cref="MaxCloudRoleInstanceValuesToDiscover"/> property.
+        /// </summary>
+        public const int MaxCloudRoleInstanceValuesToDiscoverDefault = 2;
+
+        /// <summary>
+        /// The default value for the <see cref="MaxCloudRoleNameValuesToDiscover"/> property.
+        /// </summary>
+        public const int MaxCloudRoleNameValuesToDiscoverDefault = 2;
+
+        private readonly object lockObject = new object();
+
+        /// <summary>
+        /// Extracted metric.
+        /// </summary>
+        private Metric traceDurationMetric = null;
```

Comment:
We should be extracting just the count, not the duration.

Reply:
Changed the name to traceCountMetric